File: Java/2023/test/year2023/Day12/CharacterMatcherTest.java
package year2023.Day12;

import java.util.*;

import org.junit.jupiter.api.Test;

public class CharacterMatcherTest {
    CharacterMatcher matcher = new CharacterMatcher();

    @Test
    void unitTest() {
        assert (matcher.DP("?###????????", List.of(3, 2, 1)) == 10);
        assert (matcher.heuristics("?###????????", List.of(3, 2, 1)) == 10);
        assert (matcher.DP("?#??#????.????#", List.of(8, 1, 1)) == 6);
        assert (matcher.heuristics("?#??#????.????#", List.of(8, 1, 1)) == 6);
    }
}

File: Java/2023/test/year2023/Day12/CharacterMatcherTest.kt
package year2023.Day12;

import org.testng.annotations.Test;

class CharacterMatcherTest {

    val matcher = CharacterMatcher();

    @Test
    fun unitTest() {
        assert(matcher.DP("?###????????", listOf(3, 2, 1)) == 10L);
        assert(matcher.heuristics("?###????????", listOf(3, 2, 1)) == 10L);
        assert(matcher.DP("?#??#????.????#", listOf(8, 1, 1)) == 6L);
        assert(matcher.heuristics("?#??#????.????#", listOf(8, 1, 1)) == 6L);
    }
}
